### PR TITLE
Use def span for non-ascii ident feature gate error

### PR DESCRIPTION
**Pull request by estebank**

### CODE REVIEW

**Comment by arielb1** [comment]
I'm worried this might call `def_span` on random-ish AST nodes (whatever is passed to `visit_name`), with possibly random results. What are the sort of nodes that are supposed to be valid input to `def_span`?

**Comment by estebank** [comment]
@arielb1 `def_span` is an outright hack that takes everything in the original span until the next occurrence of the char `{`, or the full span if it reaches newline before then. The case where this could be a problem would be if a user tried something like `type My{Weird}Type;`, but that would get caught by the parser.

File: src/test/ui/feature-gate-non_ascii_idents.stderr [hunk 1]
```diff
@@ -83,22 +79,16 @@ error: non-ascii idents are not fully supported. (see issue #28979)
...
+33 | enum Bär { //~ ERROR non-ascii idents
+   | ^^^^^^^^
    |
    = help: add #![feature(non_ascii_idents)] to the crate attributes to enable
 
 error: non-ascii idents are not fully supported. (see issue #28979)
   --> $DIR/feature-gate-non_ascii_idents.rs:34:5
    |
-34 | /     Bäz { //~ ERROR non-ascii idents
-35 | |         qüx: isize //~ ERROR non-ascii idents
```
Comment (bjorn3) [inline src/test/ui/feature-gate-non_ascii_idents.stderr]: Why doesn't this line give an error?

Comment (estebank) [reply src/test/ui/feature-gate-non_ascii_idents.stderr]: Good question. It wasn't already, would have to check.

**Pull request merged**